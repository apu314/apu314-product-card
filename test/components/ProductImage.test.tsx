import React from 'react'
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';



describe('ProductImage', () => {
  test('should Show the ProductImage component with the custom image', () => {
    const wrapper = renderer.create(
      <ProductImage img='./coffee-mug.png' />
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  test('should Show the ProductImage componente with the image from the product data', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {
          () => (
            <ProductImage />
          )
        }
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
