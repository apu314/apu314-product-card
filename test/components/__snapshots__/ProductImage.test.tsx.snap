// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should Show the ProductImage component with the custom image 1`] = `
<img
  alt=""
  className="productImg undefined"
  onError={[Function]}
  src="./coffee-mug.png"
/>
`;

exports[`ProductImage should Show the ProductImage componente with the image from the product data 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt=""
    className="productImg undefined"
    onError={[Function]}
    src="./coffee-mug.png"
  />
</div>
`;
